### PR TITLE
feat: adding executor tool

### PR DESCRIPTION
**Pull request by Abraxas-365**

### CODE REVIEW

**Comment by prabirshrestha** [comment]
What about arguments? Should that be an array too? Parsing ages gets quite complex when use quotes.

**Comment by prabirshrestha** [comment]
Should we also allow pre run hook? For example I'm not want to run random commands.

Few other customizations includes working dir or env vars. Or might be pre run hook could be used to set these.

**Comment by Abraxas-365** [comment]
> What about arguments? Should that be an array too? Parsing ages gets quite complex when use quotes.

Can you give an example, please? Right now, this is what the agent gives as input: [ls, mkdir test, cd test]. So, I parse it by separating the commands using the ",".

**Comment by Abraxas-365** [comment]
> Should we also allow pre run hook? For example I'm not want to run random commands.
> 
> Few other customizations includes working dir or env vars. Or might be pre run hook could be used to set these.

I think we can put that in the description of the tool, so we can create a tool.with_allow_commands

File: src/tools/command_executor/command_executor.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+#[async_trait]
+impl Tool for CommandExecutor {
+    fn name(&self) -> String {
+        String::from("GoogleSearch")
+    }
+    fn description(&self) -> String {
+        String::from(format!(
+            r#""This tool let you run command on the terminal"
+            "The input should be an array with comands for the following platform: {}"
```
Comment (prabirshrestha) [inline src/tools/command_executor/command_executor.rs:34]: you can use 

```rust
r#"line1
 line2
 line3"#);
```

you don't need `"` in each line.

File: src/tools/command_executor/command_executor.rs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                "required": ["cmd"],
+                "additionalProperties": false,
+                "description": "Object representing a command and its optional arguments"
+              }
+            }
+          },
+          "required": ["commands"],
+          "additionalProperties": false
+        }
+                )
```
Comment (prabirshrestha) [inline src/tools/command_executor/command_executor.rs:99]: indentation?

**Review by prabirshrestha — APPROVED** [review]
LGTM

**Pull request merged**